packages/editor/src/components/table-of-contents/panel.js

```diff
@@ -5,13 +5,22 @@ import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { withSelect } from '@wordpress/data';
 
+import { delay } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import WordCount from '../word-count';
 import DocumentOutline from '../document-outline';
 
-function TableOfContentsPanel( { headingCount, paragraphCount, numberOfBlocks } ) {
+function TableOfContentsPanel( { headingCount, paragraphCount, numberOfBlocks, onRequestClose } ) {
+	// Remove the block id hash from the url because block ids are only valid for the session.
+	const removeBlockIdHash = () => {
```

Comment:
Is there anything about this function which makes it specific to the rendering of this component? It seems like something which could be defined outside the component.

Part of this for me would be toward making the `onSelect` we pass to `DocumentOutline` a consistent function reference, rather than generating a new function on each render (causing React to rebind the event handler on each render). Outside of making this a `class extends Component` with an instance-bound select-handler, it's not immediately achievable, though moving it outside the component at least makes it in the direction of being an option. Frankly, the behavior of "strip hash from URL" is one which could be argued to be standalone, not specific to the behavior of this component, blocks, or an editor at all.

Reply:
> Frankly, the behavior of "strip hash from URL" is one which could be argued to be standalone, not specific to the behavior of this component, blocks, or an editor at all.

how about we build a `HashlessLink` component that wrapped a link with the hash stripping functionality. Naming suggestions?

Reply:
took a pass in https://github.com/WordPress/gutenberg/pull/10815/commits/d5470e504762d2b2f62039234ea26ea661da2de4

Reply:
went with a `removeURLHash` helper, what do you think?